keps/sig-storage/3294-provision-volumes-from-cross-namespace-snapshots/README.md

```diff
@@ -0,0 +1,1105 @@
+
+1. No `ReferenceGrant` for a `VolumeSnapshotLink` exists when the `VolumeSnapshotLink` is handled, and the `ReferenceGrant` is created later
+    A controller won't use the `VolumeSnapshot` until the `ReferenceGrant` that allows the access is created.
+
+2. `ReferenceGrant` for a `VolumeSnapshotLink` exists when the `VolumeSnapshotLink` is handled, and the `ReferenceGrant` is deleted later
+    A controller will use the `VolumeSnapshot` if the `ReferenceGrant` that allows the access exists when it checks.
+    If all the processes succeed without any error, it succeeds even the `ReferenceGrant` is deleted in the middle of the processes.
+    If any errors happened in the processes and the controller retries, it may detect that there is no `ReferenceGrant` . Then, it won't use the `VolumeSnapshot` until the `ReferenceGrant` that allows the access is re-created.
+
+#### Conflict on installing `VolumePopulator` CR for `VolumeSnapshotLink` across CSI drivers
+
+This feature requires installing `VolumePopulator` CR for `VolumeSnapshotLink` and is enabled per CSI driver basis.
+Therefore, on enabling this feature for each CSI driver, it is expected that `VolumePopulator` CR for `VolumeSnapshotLink` is created before each CSI driver installation.
+As a result, there may be a conflict in creating it for each driver, if there are any differences in their definitions, like alpha API and beta API.
+
+To avoid this issue, it should be avoided to manage `VolumePopulator` CR for `VolumeSnapshotLink` in each CSI driver's repository.
+It should be managed in another single repository and the same CR should be used per cluster basis.
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+### Example flow of how this proposal works
+
+Let's use [Story 1](#story-1) as an example and let's assume the following:
+- There are two namespaces, prod and test,
+- Alice manages the prod namespace and Bob manages the test namespace,
+- Alice would like to allow `VolumeSnapshot` foo-backup in the prod namespace to be consumed in the test namespace for testing,
+- Bob would like to create a PV for PVC foo-testing in the test namespace from the `VolumeSnapshot` foo-backup in the prod namespace.
+
+Once this proposal is implemented, it can be achieved by doing the following steps:
+
+1. In the prod namespace, Alice creates a `ReferenceGrant` bar that allows referencing to the `VolumeSnapshot` foo-backup in the prod namespace from any `VolumeSnapshotLinks` in the test namespace,
+    ```yaml
+    apiVersion: gateway.networking.k8s.io/v1alpha2
```

Comment:
Do you know if there are any efforts to make this API a core resource? 

I think it may be a little confusing for storage to depend on a networking API, so ideally this should be a more generic api group.

Reply:
I think that it’s good to move `ReferenceGrant` to more generic place if there are a lot of similar use cases.  However, we may define similar API under a storage API, if such use cases exist only in storage and network.

@robscott  @youngnick 

Any comments on this?

Reply:
Hey @ttakahashi21 and @msau42. As a general status update, we're on the verge of graduating ReferenceGrant to beta as part of the v0.6.0 release in ~3 weeks. I agree that this resource and concept are broader than networking and would ideally live somewhere that sig-auth controls. We've presented this topic to sig-arch and sig-auth previously, and at the time it was an interesting pattern but needed broader interest to justify upstreaming. It sounds like sig-storage may be willing to cosponsor this with sig-network? If so, maybe we should add this to the sig-auth agenda on Wednesday? (cc @enj, @mikedanese, @ritazh).

Reply:
https://groups.google.com/g/kubernetes-sig-auth/c/hWlkuChRFUM

Reply:
@robscott Any conclusions from the sig-auth meeting last week?

Reply:
@robscott  Thank you for raising the topic to the sig-auth meeting. I've checked the discussion in the agenda and the video. In my understanding, the API is likely to be moved. So, I've added description on "ReferenceGrant API change" in "Risks and Mitigations" section and mark it as a beta blocker.